addons/events/fnc_weaponEvents.sqf

```diff
@@ -0,0 +1,80 @@
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_weaponEvents
+
+Description:
+    Execute weapon events framework.
+
+    class MyWeapon: MyWeapon_base {
+        class EventHandlers {
+            fired = "_this call CBA_fnc_weaponEvents";
+        };
+    };
+
+Parameters:
+    _unit   - Unit that fired the weapon <OBJECT>
+    _weapon - The weapon fired by the unit <STRING>
+
+Returns:
+    Nothing
+
+Examples:
+    (begin example)
+        fired = "_this call CBA_fnc_weaponEvents";
+    (end)
+
+Author:
+    commy2
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+
+params ["_unit", "_weapon"];
+
+private _config = configFile >> "CfgWeapons" >> _weapon >> "CBA_WeaponEvents";
+
+private _isEmpty = _unit ammo _weapon == 0;
+private _onEmpty = true;
+
+if (isNumber (_config >> "onEmpty")) then {
+    _onEmpty = getNumber (_config >> "onEmpty") == 1;
+};
+
+private _fnc_soundSource = {
+    private _soundSourceName = format [QGVAR(soundSource_%1), _soundLocation];
+    private _soundSource = _unit getVariable [_soundSourceName, objNull];
+
+    if !(_soundSource in attachedObjects _unit) then {
+        _soundSource = "Building" createVehicleLocal [0,0,0];
```

Comment:
using #particlesource is faster
```Result:
0.0093 ms

Cycles:
10000/10000

Code:
JK_test pushBack ("#particlesource" createVehicleLocal [0,0,0]);
```
VS
```
Result:
0.25208 ms

Cycles:
3967/10000

Code:
JK_test pushBack ("Building" createVehicleLocal [0,0,0]);```

Reply:
How long does it live?

Reply:
until you delete it

Reply:
Perfect, thanks!

Reply:
Any reason to not use `#soundsource`? You know, because it's supposed to be a soundsource?

Reply:
```
Result:
0.0087 ms

Cycles:
10000/10000

Code:
JK_test pushBack ("#particlesource" createVehicleLocal [0,0,0]); 
```
vs 
```
Result:
0.135655 ms

Cycles:
7379/10000

Code:
JK_test pushBack ("#soundsource" createVehicleLocal [0,0,0]); 
```

Reply:
Nvm, seems like those don't work with `createVehicleLocal`.

Reply:
what does not work with createVehicleLocal?